### DISCUSSION TITLE
Custom Ubuntu 22.04 image build failing during installation of git+git ftp

### DISCUSSION THREAD

**Discussion by fajterini** [actions/virtual-environments#8863] [General questions]
### Description

Hi I am trying, to build ubuntu image in our environment, but have issue during git installation on ubuntu 2204.

Windows build is working, but I wanted to build also linux image and have this issue:

During git and git ftp installation image build is failed. 
I was looking same steps in Microsoft build at https://github.visualstudio.com/runner-images/_build/results?buildId=133985&view=logs&j=747417c8-a650-5aed-afed-d9b52dbd7d24&t=f245de8a-85db-5d11-6f6b-a3b45c2b2a6a

My build is failing just before this steps. 

==> azure-arm.build_image: # github.com:22 SSH-2.0-babeld-542480ac
==> azure-arm.build_image: # github.com:22 SSH-2.0-babeld-4a94ee6b
==> azure-arm.build_image: # github.com:22 SSH-2.0-babeld-542480ac
==> azure-arm.build_image: # ssh.dev.azure.com:22 SSH-2.0-SSHBlackbox.10

In our environment all default NSG have RDP and SSH to internet blocked by default.  
I haven't seen this communication blocked on FW, so I expect this is related to NSG.
My question is why and where is this connecting ?  I don't want to allow SSH for image build if not needed.

Edit: is this just because of this step in install script ?  images/ubuntu/scripts/build/git.sh

# Add well-known SSH host keys to known_hosts
ssh-keyscan -t rsa,ecdsa,ed25519 github.com >> /etc/ssh/ssh_known_hosts
ssh-keyscan -t rsa ssh.dev.azure.com >> /etc/ssh/ssh_known_hosts

If this is the case I will probably need to create some workaround how to fix this without opening port 22.
Is this step really needed ? I see same step is in windows build, but in my case it didn't failed or image build didn't fail.

Thanks

### Platforms affected

- [ ] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

N/A

### Is it regression?

No

### Expected behavior

Image build successful even when port 22 is blocked.

### Actual behavior

Build failed

### Repro steps

Run image build with NSG that blocks SSH connection to internet.

**Comment by mikhailkoliada** [comment]
@fajterini hey!

Yes, we are adding two well known hosts on purpose so customers do not have to do this in their workflows (lots of people are using both GitHub and ado to fetch their resources), if you do not need those you can remove the lines.

**Reply by fajterini** [reply]
Hi Mikhail,

Thanks for reply. I decided to disable this steps in our pipeline, i just commended out/replaced ssh-keyscan to # ssh-keyscan.
This is only step that require SSH port 22 to be open to internet during image build.
I understand why is there, but not how this is implemented. Other option for me would be contact our Security team and approve SSH connections to github and azure devops and I don't know how could I explain this request. They standard is that everything is blocked by default and all connections needs to be allowed if needed (whitelisted).
But probably I would still done in offline fashion where i would need to dump ssh key on other machine and import them without need of opening ports. 

And I end up like I do, It would be nice If runner images would be available in Azure market, and whole build and security would be on Microsoft :) 
For github with large runners with vnet integration we plan to use it in future, but in Azure DevOps we still need to build our images for selfhosted agents.

Thanks
